engine/src/main/java/org/terasology/engine/world/chunks/pipeline/ChunkProcessingPipeline.java

```diff
@@ -53,10 +53,19 @@
      * Create ChunkProcessingPipeline.
      */
     public ChunkProcessingPipeline(Function<Vector3ic, Chunk> chunkProvider, Flux<Chunk> chunkStream) {
+        this(chunkProvider, chunkStream, true);
+    }
+
+    /**
+     * @param threaded Whether to use worker threads to generate chunks. If it's false, all processing will be done on the main thread.
+     * It should be set to false in tests that request processing of specific chunks.
+     * Always true in a real game.
+     */
+    protected ChunkProcessingPipeline(Function<Vector3ic, Chunk> chunkProvider, Flux<Chunk> chunkStream, boolean threaded) {
         this.chunkProvider = chunkProvider;
         scheduler = Schedulers.newParallel("chunk processing", NUM_TASK_THREADS);
-        Flux<Chunk> stream = chunkStream.subscribeOn(scheduler);
-        for (int i = 0; i < NUM_TASK_THREADS; i++) {
+        Flux<Chunk> stream = threaded ? chunkStream.subscribeOn(scheduler) : chunkStream;
+        for (int i = 0; i < (threaded ? NUM_TASK_THREADS : 1); i++) {
             stream.subscribe(new BaseSubscriber<Chunk>() {
```

Comment:
this is kind of strange, seems like something just so it passes the unit test. I'm not sure this is a good direction that resolves the test?

Reply:
It just doesn't use separate threads in the pipeline tests. I agree that that's not ideal, but it's better than adding `Thread.sleep(1000)` after each `Flux.create()`, which is basically what we'd have to do otherwise to ensure that the subscription actually happens before we try to submit chunks (that was what was causing the failing test).

I guess another option would be adding another signal object for waiting until subscription happens, plus a flag to check if it's already subscribed - that would probably work, but I'm not sure it would actually be less intrusive. If you think that would be better, though, I can do that.

Reply:
just make  Schduler a parameter then you can use the FIFO scheduler.

Reply:
Good idea, I've just implemented that. Unfortunately there isn't an existing scheduler that runs everything immediately on the same thread, but a custom one that does that was just a 10 line anonymous class.